### ISSUE TITLE
source in regex works differently

### ISSUE THREAD

**Issue by sudeeptarlekar** [oracle/truffleruby#2569]
### Description
When `source` method called for regex, it yields different result than other ruby version e.g. 2.6.0. Method adds extra escape characters(`\`) for `/`

### Sample
With truffleruby
```
irb(main):001:0> /\d{4}\/\d{4,10}\/07/.source
=> "\\d{4}\\/\\d{4,10}\\/07"
irb(main):002:0>
```

With ruby 2.6.0 and other ruby versions
```
irb(main):001:0> /\d{4}\/\d{4,10}\/07/.source
=> "\\d{4}/\\d{4,10}/07"
irb(main):002:0>
```

### Difference
```
"\\d{4}\\/\\d{4,10}\\/07"   # truffleruby
"\\d{4}/\\d{4,10}/07"       # ruby 2.6.0/2.3.8/2.7.1
```

Let me know if I am doing something wrong here.

**Comment by sudeeptarlekar** [comment]
Interesting. Thanks for quick help

**Comment by eregon** [comment]
Let's keep this open as it is an observable difference.
https://github.com/faker-ruby/faker/pull/2442 looks good to me :)

**Comment by andrykonchin** [comment]
Created a ticket https://bugs.ruby-lang.org/issues/18838

**Comment by andrykonchin** [comment]
Fixed in [b30242d](https://github.com/oracle/truffleruby/commit/2d0ae5e3343ad3978abe9aab4089e5d0a29708ae), thank you for the report.
